includes/admin/class-admin.php

```diff
@@ -21,37 +21,50 @@ class Admin {
 	// Admin page title
 	protected $page_title;
 
+	// Plugin basename
+	protected $plugin_basename;
+
 	/**
 	 * Creates admin object and implements registered actions:
 	 * 1) adds option submenu page to WP Settings page
 	 * 2) sets up plugin settings and displays admin page content
+	 *
+	 * @param string $plugin_base_file The absolute full path and filename
+	 *                                 of the main plugin file
 	 */
-	protected function __construct() {
+	protected function __construct( $plugin_base_file ) {
 
 		// Setup properties used throughout this class
-		$this->page_slug   = 'hm-toc-settings';
-		$this->option_slug = 'hm_content_toc';
-		$this->page_title  = __( 'HM Content TOC Settings', 'hm-content-toc' );
+		$this->page_slug       = 'hm-toc-settings';
+		$this->option_slug     = 'hm_content_toc';
+		$this->page_title      = __( 'HM Content TOC Settings', 'hm-content-toc' );
+		$this->plugin_basename = plugin_basename( $plugin_base_file );
 
 		// Add admin submenu page to Settings
 		add_action( 'admin_menu', array( $this, 'add_plugin_option_menu_page' ) );
 
 		// Setup plugin settings and display admin page content
 		add_action( 'admin_init', array( $this, 'setup_plugin_option_settings' ) );
+
+		// Add Settings link to plugin links on main Plugin page
+		add_filter( 'plugin_action_links_' . $this->plugin_basename, array( $this, 'add_action_links' ) );
 	}
 
 	/**
 	 * Make class a singleton, as we don't need more than
 	 * one instance of it
 	 *
+	 * @param string $plugin_base_file The absolute full path and filename
+	 *                                 of the main plugin file
+	 *
 	 * @return Admin True single instance of the class
 	 */
-	public static function get_instance() {
+	public static function get_instance( $plugin_base_file ) {
```

Comment:
The `$plugin_base_file` parameter should be optional, because subsequent calls to `TOC::get_instance()` won't pass in the file name.